tmt/steps/report/junit.py

```diff
@@ -58,40 +67,229 @@ def duration_to_seconds(duration: Optional[str]) -> Optional[int]:
+        to ``custom`` value.
+    :param include_output_log: if enabled, the ``<system-out>`` tags are included
+        in the final template output.
+    :param prettify: allows to control the XML pretty print.
+    :param results_context: if set, the provided ResultsContext is used in a template.
+    :param extra_variables: if set, these variables get propagated into the
+        Jinja template.
+    """
+
+    # Get the template context for TMT results
+    results_context = results_context or ResultsContext(phase.step.plan.execute.results())
+
+    # Prepare the template environment
+    environment = default_template_environment()
+
+    template_dir = Path('steps/report/junit/templates/')
+    template_path = template_path or tmt.utils.resource_files(
+        template_dir / Path(f'{flavor}.xml.j2'))
+
+    # Use a FileSystemLoader for a non-custom flavor
+    if flavor != CUSTOM_FLAVOR_NAME:
+        environment.loader = FileSystemLoader(
+            searchpath=tmt.utils.resource_files(template_dir))
+
+    def _read_log(log: Path) -> str:
+        """ Read the contents of a given result log """
+        if not log:
+            return ''
 
-    suite = junit_xml.TestSuite(report.step.plan.name)
+        try:
+            return str(phase.step.plan.execute.read(log))
+        except tmt.utils.FileError:
+            return ''
+
+    environment.filters.update({
+        'read_log': _read_log,
+        'duration_to_seconds': duration_to_seconds,
+        'failures': tmt.result.Result.failures,
+        })
+
+    # Explicitly enable the autoescape because it's disabled by default by TMT
+    # (see /teemtee/tmt/issues/2873 for more info.
+    environment.autoescape = select_autoescape(enabled_extensions=('xml.j2'))
+
+    xml_data = render_template_file(
+        template_path,
+        environment,
+        RESULTS=results_context,
+        PLAN=phase.step.plan,
+        INCLUDE_OUTPUT_LOG=include_output_log,
+        **extra_variables)
+
+    # Try to use lxml to check the flavor XML schema and prettify the final XML
+    # output.
+    try:
+        from lxml import etree
+
+        xml_parser_kwargs: dict[str, Any] = {
+            'remove_blank_text': prettify,
+            'schema': None,
+            }
+
+        # The schema check must be done only for a non-custom JUnit flavors
+        if flavor != CUSTOM_FLAVOR_NAME:
+            xsd_schema_path = Path(tmt.utils.resource_files(
+                Path(f'steps/report/junit/schemas/{flavor}.xsd')))
+
+            schema_root: XMLElement = etree.XML(xsd_schema_path.read_bytes())
+            xml_parser_kwargs['schema'] = etree.XMLSchema(schema_root)
+        else:
+            phase.warn(
+                f"The '{CUSTOM_FLAVOR_NAME}' JUnit flavor is used, you are solely responsible "
+                "for the validity of the XML schema.")
+
+            phase.warn(f"The pretty print is always disabled for '{CUSTOM_FLAVOR_NAME}' JUnit "
```

Comment:
A warning is emitted, but I was unable to find where `prettify` is set to `False` for custom flavors.

Reply:
Good catch. I've broken the prettify behavior for custom templates when I was fixing the typing problems, I think.

It should work again:
- https://github.com/teemtee/tmt/compare/0ac068fe0636e49ec95f34eeb5a0e1f1c66de6b7..81b54df4f1340833d21fc05d2457273f4aa90edb